### PR TITLE
:bug: fix type handling in `set_up_random_state`

### LINKED ISSUE #1795: 🐛 `random_seed` command-line arg is a string but int is required
### Describe the bug

I'm having trouble using the `--random_seed` command line flag to `run.py`. The `run` command below raises the following error

```
Running BIDS validator
Traceback (most recent call last):
  File "/code/run.py", line 807, in <module>
    run_main()
  File "/code/run.py", line 593, in run_main
    'random_seed'])
  File "/code/CPAC/pipeline/random_state/seed.py", line 173, in set_up_random_state
    raise ValueError('Valid random seeds are positive integers up to '
ValueError: Valid random seeds are positive integers up to 2147483647, "random", or None, not 2022
```

Looking at the code, it seems like it might be a type issue: `--random_seed` is parsed as `str`

https://github.com/FCP-INDI/C-PAC/blob/9586d7146eff224200e18622523a56da6a9826a6/dev/docker_data/run.py#L241-L243

but `set_up_random_state` expects an `int`

https://github.com/FCP-INDI/C-PAC/blob/9586d7146eff224200e18622523a56da6a9826a6/CPAC/pipeline/random_state/seed.py#L166-L174

### To reproduce

_No response_

### Preconfig

- [X] default
- [ ] abcd-options
- [ ] anat-only
- [ ] blank
- [ ] ccs-options
- [ ] fmriprep-options
- [ ] fx-options
- [ ] monkey
- [ ] monkey-ABCD
- [ ] ndmg
- [ ] nhp-macaque
- [ ] preproc
- [ ] rbc-options
- [ ] rodent

### Custom pipeline configuration

_No response_

### Run command

```BASH
cpac --platform singularity --tag release-v1.8.4 run \
  /path/to/data \
  /path/to/outputs \
  test_config \
  --random_seed 2022
```

### Expected behavior

The `run` command would not raise an error.

### Acceptance criteria

`run` no longer raises an error when `--random_seed` is used.

### Screenshots

_No response_

### C-PAC version

v1.8.4

### Container platform

Singularity

### Docker and/or Singularity version(s)

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by clane9**
## Fixes

Fixes #1795  by @clane9

## Description

Fixes how `set_up_random_state` handles number-like `str` and `int` seed arguments.

## Tests

Check that `cpac run` with `--random_seed 42` and `--random_seed random` run without error.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the `develop` branch of the repository. 
- [x] My commit messages follow best practices.
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [x] I updated the changelog.
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by shnizzedy — APPROVED** [review]
Thanks for fixing this bug!

**Review by gkiar — APPROVED** [review]
LGTM!

**Comment by shnizzedy** [comment]
I think this change merits adding a line under a 
```Markdown
### Fixed
```
subheader [in the CHANGELOG](https://github.com/clane9/C-PAC/blob/5867210016c5df683458847ac0c8cabed3a17112/CHANGELOG.md?plain=1#L33-L34)

**Comment by clane9** [comment]
@shnizzedy ok I added a changelog entry.

File: CHANGELOG.md [hunk 1]
```diff
@@ -58,6 +58,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Relicensed C-PAC from BSD-3-Clause to LGPL-3.0-or-later
 
 ### Fixed
+- Fixed [bug](https://github.com/FCP-INDI/C-PAC/issues/1795) that was causing `cpac run` to fail when passing a manual random seed via `--random_seed`.
```
Comment (shnizzedy) [inline CHANGELOG.md:61]: Thanks for adding this. Can you please move the line up to the ["Unreleased" section](https://github.com/FCP-INDI/C-PAC/blob/b5c742a12e42b68ff78e2f039a24cdeb714216e1/CHANGELOG.md?plain=1#L17-L34)?

**Review by shnizzedy — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (clane9) [reply CHANGELOG.md]: whoops! good catch, sorry should be fixed now

**Review by shnizzedy — APPROVED** [review]

**Review by sgiavasis — APPROVED** [review]

**Pull request merged**